examples/teams_flatten.cpp

```diff
@@ -0,0 +1,162 @@
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+// Copyright (c) 2016-22, Lawrence Livermore National Security, LLC
+// and RAJA project contributors. See the RAJA/LICENSE file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+
+#include <cstdlib>
+#include <iostream>
+#include <limits>
+
+#include "memoryManager.hpp"
+
+#include "RAJA/RAJA.hpp"
+
+/*
+ *  Thread Flatten Example using RAJA Teams
+ *
+ *  This example illustrates use of the "flatten"
+ *  policy inside RAJA Teams
+ *
+ *  The flatten policy enables reshaping
+ *  multi-dimensional thread teams to 1D
+ *  simplifying indexing patterns
+ *
+ *  RAJA features shown:
+ *    - `launch' method to create kernel
+ *    -  flatten_block_threads
+ *
+ */
+
+/*
+ * Define device launch policies
+ */
+
+#if defined(RAJA_ENABLE_CUDA)
+using device_launch = RAJA::expt::LaunchPolicy<RAJA::expt::cuda_launch_t<false>>;
+using device_inner_loop0 =  RAJA::expt::LoopPolicy<RAJA::cuda_thread_x_direct>;
+using device_inner_loop1 =  RAJA::expt::LoopPolicy<RAJA::cuda_thread_y_direct>;
+using device_flatten_loop =  RAJA::expt::LoopPolicy<RAJA::expt::cuda_flatten_block_threads_xy_direct>;
+using reduce_policy = RAJA::cuda_reduce;
+#elif defined(RAJA_ENABLE_HIP)
+using device_launch = RAJA::expt::LaunchPolicy<RAJA::expt::hip_launch_t<false>>;
+using device_inner_loop0 =  RAJA::expt::LoopPolicy<RAJA::hip_thread_x_direct>;
+using device_inner_loop1 =  RAJA::expt::LoopPolicy<RAJA::hip_thread_y_direct>;
+using device_flatten_loop =  RAJA::expt::LoopPolicy<RAJA::expt::hip_flatten_block_threads_xy_direct>;
```

Comment:
TODO: change _loop -> _pol to avoid confusion from internal RAJA policies (_loop vs _direct)